docs/dev-guides/PERSONAL-ENVIRONMENTS.md

```diff
@@ -0,0 +1,523 @@
+### The Helmfile
+The helmfile allows specifying override values for a chart for a particular
+deployment, and the [terra-helmfile repo](https://github.com/broadinstitute/terra-helmfile)
+is the place providing and for editing those. This "gitOps" model
+is similar to the relationship between the private, Atlantis-enabled `terraform-ap-deployments` repo
+and the public `terraform-ap-modules` that we saw above.
+
+We don't actually need `helm` installed locally for this step, though it will be required later.
+
+
+Clone the [terra-helmfile](https://github.com/broadinstitute/terra-helmfile) and look at the file structure.
+```shell script
+$ tree -L 2
+```
+
+```
+.
+├── README.md
+├── argocd
+│   └── values
+├── bin
+│   ├── logging.sh
+│   └── render
+├── clusters.yaml
+├── docs
+│   └── NAMING.md
+├── envDefaults.yaml
+├── environments
+│   ├── live
+│   ├── live.yaml
+│   ├── personal
+│   ├── personal.yaml
+│   └── preview.yaml
+├── helmfile.yaml
+├── terra
+│   └── values
+├── versions
+│   ├── README.md
+│   ├── alpha.yaml
+│   ├── prod.yaml
+│   └── staging.yaml
+└── versions.yaml
+
+```
+You can see the main `helmfile.yaml`, an `environments` directory with subdirectories for `live` and `personal`, and a `terra` directory
+with subdirectories for each service:
+```shell script
+jaycarlton~/repos/terra-helmfile/terra (master) $ tree -L 2
+```
+
+```
+.
+└── values
+    ├── agora
+    ├── buffer
+    ├── buffer.yaml
+    ├── consent
+    ├── crljanitor
+    ├── crljanitor.yaml
+    ├── cromiam
+    ├── cromwell
+    ├── datarepo
+    ├── datarepomonitoring
+    ├── duos
+    ├── firecloudorch
+    ├── global
+    ├── global.yaml
+    ├── hydra
+    ├── icdemo
+    ├── identityconcentrator
+    ├── istioconfig
+    ├── istioconfig.yaml
+    ├── leonardo
+    ├── ontology
+    ├── poc
+    ├── poc.yaml
+    ├── rawls
+    ├── sam
+    ├── workspacemanager
+    └── workspacemanager.yaml
+```
+For this PR we'll edit all of
+* the helmfile itself, to make it aware of our personal enviornment
+* a personal.yaml file defining our environment's enabled services at a  high level.
+* several values files defining parameters for the individual services in the environment
+
+### Helmfile
+The main helmfile itself gets a simple two-line entry for this project pointing to other files with
+properties to read (including the one above):
+```yaml
+  jcarlton:
+    values:
+      - environments/personal.yaml
+      - environments/personal/jcarlton.yaml
+```
+
+### Personal yaml file
+This file describes the Terra apps that are required for the personal environment. In my case those were
+resource buffer, janitor, and workspace manager. For the file `terra-helmfile/environments/personal/jcarlton.yaml`:
+```yaml
+# Environment overrides for jcarlton environment
+releases:
+  crljanitor:
+    enabled: true
+  workspacemanager:
+    enabled: true
+  buffer:
+    enabled: true
+```
+If any environments are disabled, their resources are not generated by Terraform.
```

Comment:
s/environments/applications/

And note, the Helmfile does not have any bearing on resources getting generated by Terraform. I think we need to manually to keep the Terraform app-enablement in sync with the Helmfile app-enablement.

Reply:
good catch